pkg/downloader/manager.go

```diff
@@ -220,12 +294,19 @@ func (m *Manager) downloadAll(deps []*chartutil.Dependency) error {
 			if m.Debug {
 				fmt.Fprintf(m.Out, "Archiving %s from repo %s\n", dep.Name, dep.Repository)
 			}
-			ver, err := tarFromLocalDir(m.ChartPath, dep.Name, dep.Repository, dep.Version)
+			ver, archiveFile, err := tarFromLocalDir(m.ChartPath, dep.Name, dep.Repository, dep.Version)
 			if err != nil {
 				saveError = err
 				break
 			}
 			dep.Version = ver
+			if m.Recursive {
+				fmt.Fprintln(m.Out, "Unpacking:", archiveFile)
+				err := chartutil.ExpandFile(destPath, archiveFile)
```

Comment:
I think, it's a good idea to flatten the tree but then there is the limitation which you just described. 

Maybe after building the tree, it can be traversed again to delete all the source files and just keep the `Charts` with the archive files. Something like:
```
alpine-0.1.0.tgz
charts/
  wordpress-2.1.0.tgz
  mariadb-4.3.1.tgz
```